### PR TITLE
feat: add Azure e2e testenv

### PR DESCRIPTION
**Pull request by akijakya**
## Description

Run e2e tests on Azure provider. 
Some things needed to be adjusted on the e2e tests side so they run successfully on Azure.

## Type of Change

[ ] Bug Fix  
[X] New Feature  
[ ] Breaking Change  
[ ] Refactor  
[ ] Documentation  
[ ] Other (please describe)  

## Checklist

- [X] I have read the [contributing guidelines](https://github.com/openclarity/vmclarity/blob/main/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [X] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [x] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

File: testenv/azure/asset.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+func (e *AzureEnv) createSubnet(ctx context.Context, resourceGroupName string) (*armnetwork.Subnet, error) {
+	pollerResp, err := e.subnetsClient.BeginCreateOrUpdate(
+		ctx,
+		resourceGroupName,
+		virtualNetworkName,
+		subnetName,
+		armnetwork.Subnet{
+			Properties: &armnetwork.SubnetPropertiesFormat{
+				AddressPrefix: to.Ptr("10.1.0.0/24"),
```
Comment (ramizpolic) [inline testenv/azure/asset.go:94]: can be moved to `networkSubnetAddress` const

File: testenv/azure/asset.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+					},
+				},
+			},
+			HardwareProfile: &armcompute.HardwareProfile{
+				VMSize: to.Ptr(armcompute.VirtualMachineSizeTypes("Standard_B1ls")), // VM size include vCPUs,RAM,Data Disks,Temp storage.
+			},
+			OSProfile: &armcompute.OSProfile{ //
+				ComputerName:  to.Ptr("vmclarity-asset-compute"),
+				AdminUsername: to.Ptr("ubuntu"),
+				AdminPassword: to.Ptr("Password01!@#"),
```
Comment (ramizpolic) [inline testenv/azure/asset.go:175]: should this be specified?

File: testenv/azure/asset.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+		},
+		Properties: &armcompute.VirtualMachineProperties{
+			StorageProfile: &armcompute.StorageProfile{
+				ImageReference: &armcompute.ImageReference{
+					// search image reference
+					// az vm image list --output table
+					Offer:     to.Ptr("0001-com-ubuntu-minimal-focal"),
+					Publisher: to.Ptr("Canonical"),
+					SKU:       to.Ptr("minimal-20_04-lts"),
+					Version:   to.Ptr("20.04.202004230"),
```
Comment (ramizpolic) [inline testenv/azure/asset.go:159]: lets move this out to struct var

File: testenv/azure/asset.go [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+				OSDisk: &armcompute.OSDisk{
+					Name:         to.Ptr(osDiskName),
+					CreateOption: to.Ptr(armcompute.DiskCreateOptionTypesFromImage),
+					ManagedDisk: &armcompute.ManagedDiskParameters{
+						StorageAccountType: to.Ptr(armcompute.StorageAccountTypesStandardSSDLRS), // OSDisk type Standard/Premium HDD/SSD
+					},
+				},
+			},
+			HardwareProfile: &armcompute.HardwareProfile{
+				VMSize: to.Ptr(armcompute.VirtualMachineSizeTypes("Standard_B1ls")), // VM size include vCPUs,RAM,Data Disks,Temp storage.
```
Comment (ramizpolic) [inline testenv/azure/asset.go:170]: lets also move this to a const

File: testenv/azure/utils.go [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+
+	if len(missingEnvs) > 0 {
+		return fmt.Errorf("missing environment variables: %s", strings.Join(missingEnvs, ", "))
+	}
+
+	return nil
+}
+
+func (e *AzureEnv) createAzureResources(ctx context.Context) error {
+	template, err := readTemplateJSON("./../installation/azure/vmclarity.json")
```
Comment (ramizpolic) [inline testenv/azure/utils.go:64]: lets use existing logic https://github.com/openclarity/vmclarity/blob/54dac0961ed4675e01f24f50e47ecba83684e26f/installation/manifests.go#L33C6-L33C7

File: testenv/azure/azure.go [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+	"github.com/openclarity/vmclarity/testenv/types"
+	"github.com/openclarity/vmclarity/testenv/utils"
+	"github.com/openclarity/vmclarity/testenv/utils/docker"
+)
+
+const (
+	defaultRemoteUser                          = "vmclarity"
+	bicepFileResourcesPrefix                   = "vmclarity-"
+	bicepFileDiscoverRoleDeploymentNamePostfix = "-discover-role"
+	vmClarityServerPublicIPName                = bicepFileResourcesPrefix + "server-public-ip"
```
Comment (ramizpolic) [inline testenv/azure/azure.go:42]: lets use format definitions instead. also can be moved to files where these values are used (if not here then in azure utils)

```go
bicepFileResourcesFormat = "vmclarity-%s"
bicepFileDiscoverRoleDeploymentNameFormat = "%s-discover-role"
vmClarityServerPublicIPName = "vmclarity-server-public-ip"
```

File: testenv/azure/asset.go [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+	pollerResp, err := e.virtualNetworksClient.BeginCreateOrUpdate(
+		ctx,
+		resourceGroupName,
+		virtualNetworkName,
+		armnetwork.VirtualNetwork{
+			Location: to.Ptr(location),
+			Properties: &armnetwork.VirtualNetworkPropertiesFormat{
+				AddressSpace: &armnetwork.AddressSpace{
+					AddressPrefixes: []*string{
+						to.Ptr("10.1.0.0/16"),
```
Comment (ramizpolic) [inline testenv/azure/asset.go:69]: can be moved to const `virtualNetworkAddress`

File: e2e/helpers.go [hunk 8]
```diff
@@ -94,7 +99,18 @@ func UpdateScanConfigToStartNow(config *apitypes.ScanConfig) *apitypes.ScanConfi
 		},
 		Scheduled: &apitypes.RuntimeScheduleScanConfig{
 			CronLine:      config.Scheduled.CronLine,
-			OperationTime: to.Ptr(time.Now()),
+			OperationTime: to.Ptr(time.Now().Add(5 * time.Second)), // nolint:gomnd
```
Comment (ramizpolic) [inline e2e/helpers.go:102]: add docs why this is added and use `scanStartOffset` var

File: e2e/helpers.go [hunk 9]
```diff
@@ -94,7 +99,18 @@ func UpdateScanConfigToStartNow(config *apitypes.ScanConfig) *apitypes.ScanConfi
...
 		},
 	}
 }
+
+// "database is locked" errors from SQLite should be ignored during the test
+// they occur frequently on Azure, but they are temporary and should not cause the test to fail.
+func skipDBLockedErr(err error) error {
```
Comment (ramizpolic) [inline e2e/helpers.go:109]: this should be removed in the future. lets have a ticket for this

**Review by ramizpolic — CHANGES_REQUESTED** [review]

File: installation/azure/vmclarity-install.sh [hunk 10]
```diff
@@ -87,6 +87,9 @@ systemctl daemon-reload
 # Enable and start/restart VMClarity backend
 systemctl enable vmclarity.service
 systemctl restart vmclarity.service
+
+# Add admin user to docker group and activate the changes
+usermod -a -G docker __AdminUsername__
```
[the remark this answers is not in the document]
Comment (akijakya) [reply installation/azure/vmclarity-install.sh:92]: yes, for us to be able to get docker logs through an SSH tunnel!

File: installation/azure/vmclarityDeployModule.bicep [hunk 11]
```diff
@@ -143,6 +143,7 @@ var params = {
   ExternalDBName: externalDBName
   ExternalDBUsername: externalDBUsername
   ExternalDBPassword: externalDBPassword
+  AdminUsername: adminUsername
```
[the remark this answers is not in the document]
Comment (akijakya) [reply installation/azure/vmclarityDeployModule.bicep:146]: yes, in the Azure installation template, the username has to be provided by the user, so we cannot use a hardcoded `vmclarity` value for example, but need to pass the provided username to the deploy script in which it is used to add the user to the docker group!

File: testenv/azure/asset.go (hunk 2 above)
Comment (akijakya) [reply testenv/azure/asset.go:182]: These are required fields I'm afraid...

File: e2e/helpers.go (hunk 9 above)
Comment (akijakya) [reply e2e/helpers.go:110]: https://github.com/openclarity/vmclarity/issues/1654

File: testenv/azure/asset.go (hunk 2 above)
Comment (ramizpolic) [reply testenv/azure/asset.go:182]: we should be able to do it without having to specify the password

File: e2e/helpers.go [hunk 12]
```diff
@@ -16,15 +16,21 @@
...
 
 	apitypes "github.com/openclarity/vmclarity/api/types"
 	"github.com/openclarity/vmclarity/core/to"
 )
 
-const DefaultTimeout = 60 * time.Second
+const (
+	defaultTimeout  = 5 * time.Minute
+	defaultPeriod   = 5 * time.Second
+	scanStartOffset = 5 * time.Second
```
Comment (ramizpolic) [inline e2e/helpers.go:32]:
```suggestion
	DefaultTimeout  = 5 * time.Minute
	DefaultPeriod   = 5 * time.Second
	
	fullScanStartOffset = 5 * time.Second
```

File: testenv/azure/asset.go (hunk 2 above)
Comment (akijakya) [reply testenv/azure/asset.go:182]: yes...

Comment (akijakya) [reply testenv/azure/asset.go:182]: This asset VM doesn't have a public IP though

File: e2e/helpers.go (hunk 12 above)
Comment (akijakya) [reply e2e/helpers.go:32]: Why do the first two constants need to be exported?

Comment (ramizpolic) [reply e2e/helpers.go:32]: they dont, its just that they are reused across e2e tests and it makes it clearer for usage

Comment (akijakya) [reply e2e/helpers.go]: ok, done!

File: testenv/azure/asset.go (hunk 2 above)
Comment (ramizpolic) [reply testenv/azure/asset.go:182]:
```go
			OSProfile: &armcompute.OSProfile{ // use username/password
				ComputerName:  to.Ptr("vmclarity-asset-compute"),
				AdminUsername: to.Ptr("vmclarity"),
				LinuxConfiguration: &armcompute.LinuxConfiguration{
					DisablePasswordAuthentication: to.Ptr(true),
				},
			},
```

Comment (akijakya) [reply testenv/azure/asset.go]: I added ssh auth instead of username/password!

Comment (ramizpolic) [reply testenv/azure/asset.go]: afaik we dont even need to be able to log into the VM since we are just going to create a snapshot and run the scan on it

**Review by ramizpolic — APPROVED** [review]
LGTM, thanks @akijakya! 🚀

**Pull request merged**